katz/src/main/kotlin/katz/typeclasses/Functor.kt

```diff
@@ -17,5 +17,11 @@
 package katz
 
 interface Functor<F> {
+
     fun <A, B> map(fa: HK<F, A>, f: (A) -> B): HK<F, B>
+
+    fun <A, B> lift(f: (A) -> B): (HK<F, A>) -> HK<F, B> =
```

Comment:
I haven't checked the rest of the repository, so same as above in general I'm weary of default implementations on interfaces because they enable mixin-like composition which I find can lead to messy composition.

Maybe it's just personal bias based off gut feel, but as long as we have an easy replacement in extension functions I would push for it. Drawback: interop with Java would be with statics, but AFAIK that's not this project's concern.